server/src/internalClusterTest/java/org/opensearch/indices/IndicesRequestCacheDiskTierIT.java

```diff
@@ -111,7 +113,257 @@ public void testDiskTierStats() throws Exception {
+    public void testDiskTierInvalidationByCacheCleanerPartOfDiskTier() throws Exception {
+        int thresholdInMillis = 4_000;
+        Instant start = Instant.now();
+        int heapSizeBytes = 987;
+        String node = internalCluster().startNode(
+            Settings.builder()
+                .put(IndicesService.INDICES_REQUEST_CACHE_DISK_CLEAN_INTERVAL_SETTING.getKey(), TimeValue.timeValueMillis(thresholdInMillis))
+                .put(IndicesService.INDICES_REQUEST_CACHE_DISK_CLEAN_THRESHOLD_SETTING.getKey(), "1%")
+                .put(IndicesRequestCache.INDICES_CACHE_QUERY_SIZE.getKey(), new ByteSizeValue(heapSizeBytes))
+                .put(DiskTierTookTimePolicy.DISK_TOOKTIME_THRESHOLD_SETTING.getKey(), TimeValue.ZERO) // allow into disk cache regardless of took time
+        );
+        Client client = client(node);
+
+        Settings.Builder indicesSettingBuilder = Settings.builder()
+            .put(IndicesRequestCache.INDEX_CACHE_REQUEST_ENABLED_SETTING.getKey(), true)
+            .put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, 1)
+            .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 0);
+
+        assertAcked(
+            client.admin().indices().prepareCreate("index").setMapping("k", "type=text").setSettings(indicesSettingBuilder).get()
+        );
+
+        indexRandom(true, client.prepareIndex("index").setSource("k", "hello"));
+        ensureSearchable("index");
+        SearchResponse resp;
+
+        resp = client.prepareSearch("index").setRequestCache(true).setQuery(QueryBuilders.termQuery("k", "hello" + 0)).get();
+        int requestSize = (int) getCacheSizeBytes(client, "index", TierType.ON_HEAP);
+        assertTrue(heapSizeBytes > requestSize);
+
+        int numOnDisk = 2;
+        int numRequests = heapSizeBytes / requestSize + numOnDisk;
+        for (int i = 1; i < numRequests; i++) {
+            resp = client.prepareSearch("index").setRequestCache(true).setQuery(QueryBuilders.termQuery("k", "hello" + i)).get();
+            assertSearchResponse(resp);
+            IndicesRequestCacheIT.assertCacheState(client, "index", 0, i + 1, TierType.ON_HEAP, false);
+            IndicesRequestCacheIT.assertCacheState(client, "index", 0, i + 1, TierType.DISK, false);
+        }
+
+        RequestCacheStats requestCacheStats = client.admin()
+            .indices()
+            .prepareStats("index")
+            .setRequestCache(true)
+            .get()
+            .getTotal()
+            .getRequestCache();
+
+        // make sure we have 2 entries in disk.
+        long entries = requestCacheStats.getEntries(TierType.DISK);
+        assertEquals(2, entries);
+
+        // force refresh so that it creates stale keys in the cache for the cache cleaner to pick up.
+        flushAndRefresh("index");
+        client().prepareIndex("index").setId("1").setSource("k", "good bye");
+        ensureSearchable("index");
+
+        for (int i = 0; i < 6; i++) { // index 5 items with the new readerCacheKeyId
```

Comment:
Isn't this 6 items?

Reply:
Yep, i'm considering going back to kinder garden with my son